Fb pr gating (#14)
microsoft/GHAzDO-Resources: src/pr-gating/Setup.md

  Save the PAT key before closing the UI.

- Add CIVerify as a new pipeline.
  
  Pipelines - New Pipeline - Select your code location and configure your pipeline as an existing yaml file.

  Select the CIVerify.yml file that was added in step 1.  Save the new Pipeline.

  Review the .yaml file, open up the UI to add a new Variable.

  Add a variable called GATING_PAT and set it's content to the PAT key you got in the previus step.

  Save your pipeline and optionaly run it.

  Find your new pipeline in the ADO UI, [rename it](https://learn.microsoft.com/en-us/azure/devops/pipelines/customize-pipeline?view=azure-devops#pipeline-settings) to CIVerify.

- Setup [build verification for your main branch](https://learn.microsoft.com/en-us/azure/devops/repos/git/branch-policies?view=azure-devops&tabs=browser#build-validation). Pick the CIVerify pipeline, you can keep the rest of the default settings.

<img width="800" alt="buildpolicy" src="https://github.com/microsoft/GHAzDO-Resources/assets/106392052/74801e80-46e1-4d05-97b1-5f11396330e1">

- Figure out a way to test your new setup without adding any bad code to your main branch :) 

## Possible customizations

> - Currently both language (javascript) and query suites (security-extended) are hard coded in CIVerify.yml. These settings might have to be updated to match your needs. 

Reviewer: ```suggestion
- The CodeQL language (javascript) is hardcoded in CIVerify.yml. GHAzDO settings will need to be updated to match your needs. 
```

> - The CI Verification could be extended to also (or only) check new Dependency alerts. The same strategy could be used but tweaks would have to be done to both CIVerify.yml and CIGate.ps1. 

Reviewer: - [ ] TODO - validate this is possible

Author: Checked and yes. They will not show up in the branch filter for dependencies. Still, they are accessible via the API and same technique can be used.

Reviewer: Confirmed the same:

```
S C:\repos\microsoft\GHAzDO-Resources> . 'C:\repos\microsoft\GHAzDO-Resources\src\pr-gating\CIGate.ps1'
Will check to see if there are any new  issues in this PR branch
PR source : refs/pull/1/merge. PR target: refs/heads/master
##[error] There are more alerts in the PR source branch refs/pull/1/merge, compared to target branch refs/heads/master :

##vso[task.logissue type=error;sourcepath=WebGoat.NET/Views/Shared/_LoginPartial.cshtml WebGoat.NET/Views/Shared/_LoginPartial.cshtml;linenumber=20 20;columnnumber=113 113] New  issue detected #28 : Cross-site scripting (cs/web/xss).
##[error] Fix or dismiss this new alert in the Advanced Security UI for pr branch refs/pull/1/merge.
##[error] Details for this new alert:  https://dev.azure.com/octodemo-felickz/_git/proj-felickz/alerts/28?branch=refs/pull/1/merge

##vso[task.logissue type=error;sourcepath=s/WebGoat.NET/WebGoat.NET.csproj;linenumber=;columnnumber=] New  issue detected #152 : Cookie parsing failure (GHSA-hxrm-9w7p-39cc).
##[error] Fix or dismiss this new alert in the Advanced Security UI for pr branch refs/pull/1/merge.
##[error] Details for this new alert:  https://dev.azure.com/octodemo-felickz/_git/proj-felickz/alerts/152?branch=refs/pull/1/merge

##[error] Please review these Code Scanning alerts for the  branch using the regular Advanced Security UI
##[error] Dissmiss or fix the issues listed and try re-queue the CIVerify task.
```

Reviewer: Added 
[Specify no alert type (code and dependency alerts)](https://github.com/microsoft/GHAzDO-Resources/pull/14/commits/ecfefc7104b8f83d1162d6394a4bd6aa5931a465)

One complication to concider if you want to use the same strategy for dependency scanning. It could be that the reason we see new issues reported is that the advisory database has been extended since the last scan of main. That is, the new additions in the PR is not the issue but rather that this scan is the first one after an extention of the known vunarabilities. Still, since the alerts can be dismissed in the PR branch, or a new scan can be run on the main branch, I think this is acceptable. 

> - Currently it is hard coded that we compaire to the main branch. If you have a different merge strategy, CIVerify.yml and CIGate.ps1 can be updated to accommodate that. 

Reviewer: specifying params for the `base`/`head` branches would be  💯  - default them to the default branch and the current merge branch?

Author: This was an easy change. I also updated the documentation to make it clear that this check is generic.
